src/utils/__tests__/__snapshots__/style-utils.spec.js

```diff
@@ -0,0 +1,59 @@
+import { setTextFontSize } from '../../style-utils';
+
+describe('setTextFontSize function', () => {
+  let text;
+  let font;
+  let textFontSize;
+  let hasIcon;
+
+  beforeEach(() => {
+    text = {
+      style: { fontSize: '12px' },
+      offsetWidth: '20',
+      children: new Proxy(
+        [
+          {
+            style: { marginRight: '0px' },
+          },
+          {
+            style: { marginRight: '0px' },
+          },
+        ],
+        {
+          set: () => true,
+        }
+      ),
+    };
+    font = { style: { italic: true } };
+    textFontSize = 12;
+    hasIcon = true;
+  });
+
+  it('should set font size and margins for italic font with icon', () => {
+    setTextFontSize(text, font, textFontSize, hasIcon);
+    expect(text.style.fontSize).toBe('10.08px');
+    expect(text.children[0].style.marginRight).toBe('0.8px');
+    expect(text.children[1].style.marginRight).toBe('0.8px');
+  });
+
+  it('should set font size and margins for italic font without icon', () => {
+    hasIcon = false;
+    setTextFontSize(text, font, textFontSize, hasIcon);
+    expect(text.style.fontSize).toBe('10.8px');
+    expect(text.children[0].style.marginRight).toBe('0.4px');
+  });
+
+  it('should set font size and margins for non-italic font with icon', () => {
+    font.style.italic = false;
+    setTextFontSize(text, font, textFontSize, hasIcon);
+    expect(text.style.fontSize).toBe('10.56px');
+    expect(text.children[0].style.marginRight).toBe('0.8px');
+  });
+
+  it('should set font size for non-italic font without icon', () => {
+    hasIcon = false;
+    font.style.italic = false;
+    setTextFontSize(text, font, textFontSize, hasIcon);
+    expect(text.style.fontSize).toBe('11.040000000000001px');
```

Comment:
This seems not correct, I would suggest incorporating `parseFloat(CALCULATED_FONT_SIZE).toFixed(2)` somewhere in `setTextFontSize` function.

Reply:
something like this: 
```ts
export const setTextFontSize = (text, font, textFontSize, hasIcon) => {
  let fontSize;
  if (font.style && font.style.italic) {
    if (hasIcon) {
      fontSize = Math.max(textFontSize * 0.84, 8);
      text.children[0].style.marginRight = `${text.offsetWidth * 0.04}px`;
      text.children[1].style.marginRight = `${text.offsetWidth * 0.04}px`;
    } else {
      fontSize = Math.max(textFontSize * 0.9, 8);
      text.children[0].style.marginRight = `${text.offsetWidth * 0.02}px`;
    }
  } else if (hasIcon) {
    fontSize = Math.max(textFontSize * 0.88, 8);
    text.children[0].style.marginRight = `${text.offsetWidth * 0.04}px`;
  } else {
    fontSize = Math.max(textFontSize * 0.92, 8);
  }
  
  text.style.fontSize = `${parseFloat(fontSize).toFixed(2)}px`;
};
```

haven't tried it out since i don't all project glued together, but make sure to give it a try first!